### DISCUSSION TITLE
为什么Select组件不给一个默认选中第一项的api

### DISCUSSION THREAD

**Discussion by ByYogi** [ant-design/ant-design#42047] [Q&A] [answered]
在动态获取数据情况下，这种需求很明显的吧。
现在的解决方案就是，将key改成动态的与数据statedata关联。然后defaultValue={statedata[0]}重新渲染，感觉key改成动态的很粗暴

**Comment by zombieJ** [comment, marked as answer]
你问到了React的哲学问题，可以看一下官方文档关于状态的说明：https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components

**Reply by ByYogi** [reply]
大概理解是: 组件设计尽量不要是”受控组件(controlled)“。不然用起来会很复杂

**Reply by vaynevayne** [reply]
哥, 非受控模式, 你都无法回显数据
